modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java

```diff
@@ -269,6 +271,51 @@ private Sku _toSku(Long cpInstanceId) throws Exception {
 				cpInstanceId, contextAcceptLanguage.getPreferredLocale()));
 	}
 
+	private Sku _updateNestedResources(Sku sku, CPInstance cpInstance)
+		throws Exception {
+
+		SkuSubscriptionConfiguration skuSubscriptionConfiguration =
+			sku.getSubscriptionConfiguration();
+
+		if (skuSubscriptionConfiguration != null) {
+			String subscriptionTypeValue = null;
+
+			SkuSubscriptionConfiguration.SubscriptionType subscriptionType =
+				skuSubscriptionConfiguration.getSubscriptionType();
+
+			if (subscriptionType != null) {
+				subscriptionTypeValue = subscriptionType.getValue();
+			}
+
+			_cpInstanceService.updateSubscriptionInfo(
+				cpInstance.getCPInstanceId(),
+				GetterUtil.get(
+					skuSubscriptionConfiguration.getOverrideSubscriptionInfo(),
+					cpInstance.isOverrideSubscriptionInfo()),
+				GetterUtil.get(
+					skuSubscriptionConfiguration.getEnable(),
+					cpInstance.isSubscriptionEnabled()),
+				GetterUtil.get(
+					skuSubscriptionConfiguration.getLength(),
+					cpInstance.getSubscriptionLength()),
+				subscriptionTypeValue,
+				UnicodePropertiesBuilder.create(
+					skuSubscriptionConfiguration.getSubscriptionTypeSettings(),
+					true
+				).build(),
+				GetterUtil.get(
+					skuSubscriptionConfiguration.getNumberOfLength(),
+					cpInstance.getMaxSubscriptionCycles()),
```

Comment:
this is not right, in the acceptance criteria it's written So that I can do what can be done from the UI in a headless Scenario, with this code is not possible to set the delivery subscription infos